src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -83,6 +85,7 @@ private void CheckStart()
                 currentStatus = _status;
                 if (currentStatus == Free)
                 {
+                    _timeoutOrCancellationSource ??= new();
```

Comment:
You should be able to mark this method as `[MemberNotNull(nameof(_timeoutOrCancellationSource))]`, which should let you remove at least one if not more `!`s on use of that field.

Reply:
Do you mean other methods? This is the one that lazily initializes the `CancellationTokenSource` (`Finish` then nulls it out if it has become canceled).

Reply:
I mean putting that attribute on this method should let you remove the `!` from other methods that call this one.